ssl/statem/extensions_srvr.c

```diff
@@ -137,6 +137,30 @@ int tls_parse_ctos_server_name(SSL *s, PACKET *pkt, unsigned int context,
     return 1;
 }
 
+int tls_parse_ctos_maxfragmentlen(SSL *s, PACKET *pkt, unsigned int context,
+                                  X509 *x, size_t chainidx, int *al)
```

Comment:
Shouldn't we handle the situation where s->hit is true but the value sent by the client does not match that in the SSL_SESSION? The spec isn't too clear what to do in various scenarios. I'd think: if s->hit is true and its in SSL_SESSION but the client didn't send the same value in the ClientHello (or the extension is missing altogether) then its an illegal parameter. If s->hit is true and we didn't originally negotiate it, but its in the ClientHello now then....actually I'm not sure what we do then.